File: skipad-collector/src/com/skipad/collector/common/IAd.java
package com.skipad.collector.common;

import java.util.List;

import org.json.JSONException;
import org.json.JSONObject;
import org.json.JSONTokener;

import com.skipad.collector.vastElements.VastElement;

public interface IAd {
	long getId();
    String getName();
    String getAuid();
    IResource getVideo();
    IResource getImage();
    List<IResource> getResources();
    VastElement getVast(String cid, IPackage propertiesPackage, String version);
    public Integer getPackageId();
    
    public static class ColumnNames
    {
        public static final String ModifyDate = "ModifyDate";
        public static final String AdId = "Ads_Id";
        public static final String AdName = "AdName";
        public static final String Auid = "Auid";
        public static final String CampaignId = "CampaignId";
        public static final String CampaignName = "CampaignName";
        public static final String Description = "AdDesc";
		public static final String PackageId = "AccountProperitesPackageId";
		public static final String SurveyUrl = "SurveyURL";
		public static final String ExternalTagUrl = "ExternalTagURL";
		public static final String SkipTypeId = "SkipTypeId";
    }

	JSONObject getMraid(String cid, IPackage propertiesPackage, String source) throws JSONException;
	VastElement getVpaid(String cid, IPackage propertiesPackage, String version) throws JSONException;
	VastElement getVpaidFlash(String cid, IPackage propertiesPackage, String rawParams, String version);
}

File: Skipad.Common/IAd.cs
using System;
using System.Collections.Generic;
using Jayrock.Json;

namespace Inqwise.Skipad.Common
{
    public interface IAd : IEnumerable<IResource>
    {
        long Id { get; }
        string Name { get; }
        Guid? Auid { get; }
        int CampaignId { get; }
        IResource Video { get; }
        IResource Image { get; }
        string Description { get; }
        DateTime ModifyDate { get; }
        JsonObject ToJson();
        IEnumerable<IResource> Resources { get; }
        string CampaignName { get; }
        string ExternalTagUrl { get; }
        string SurveyUrl { get; }
        long Copy(string name, string username);
        AdType AdType { get; }
        IResource Audio { get; }
        bool IsActive { get; }
    }
}